### PR TITLE
Fixes #21424 - use factory_bot_rails

### PR DESCRIPTION
**Pull request by ares**
---

as a reproducer, see http://ci.theforeman.org/job/test_plugin_matrix/3949/database=postgresql,ruby=2.4,slave=fast/console for example

### CODE REVIEW

**Comment by timogoebel** [comment]
👎 for them changing the name of the gem.

@ares: Thanks for fixing this, but I think `factory_girl 4.8.2` (that included the rename) was yanked from rubygems.org. So tests should be fine for now, I believe.

**Comment by tbrisker** [comment]
@ares thanks, I think you also need to s/FactoryGirl/FactoryBot everywhere.

**Comment by ares** [comment]
Yeah, I'll rename the rest, we can hold on until they release a new version but I suppose we want to use newer versions of the gem regardless of its name. Let's keep the discussion about renaming it out of Foreman. If anyone wants to participate in it - https://github.com/thoughtbot/factory_bot/issues/921

**Comment by ekohl** [comment]
[test foreman]

**Comment by ares** [comment]
it would be good to get this in soon, it's getting out of date with every PR that adds new tests with FactoryGirl

**Comment by timogoebel** [comment]
[test katello]

File: test/factories/host_related.rb
Comment (tbrisker) [reply test/factories/host_related.rb]: I know this is outside the scope, but maybe take advantage of this pr and fix these two?

**Comment by tbrisker** [comment]
I suggest sending a warning to foreman-dev list about this asking all plugin maintainers to prepare PRs with `s/FactoryGirl/FactoryBot/g`, and merge a couple of days after that. I opened https://github.com/theforeman/foreman/pull/4948 to unblock CI for now.

**Comment by lzap** [comment]
How about an alias class and a ticket to deprecate it in 1.18?

**Comment by tbrisker** [comment]
Sent out a mail to foreman-dev giving heads up. 
@ares - Please rebase.
@theforeman/core - please **DO NOT MERGE** until November 1st.

**Comment by ares** [comment]
I'll rebase on Nov 1, every PR that adds a test using FG breaks this.

**Comment by ares** [comment]
@tbrisker I pushed the new version, I'd kindly ask to not merge other PRs before this gets in. Could you please take a look before tests are finished so we could start the merge chain ASAP? Meanwhile I'll go over plugins and will send out PR if I don't see them opened.

**Comment by tbrisker** [comment]
Ack, waiting for tests to pass and then i'll merge.

**Comment by tbrisker** [comment]
@ares - looks like you missed a couple:
```
2.4.0@foreman_gemset ~/foreman (ares-fix/21424) $ git grep -i girl
test/controllers/api/v2/domains_controller_test.rb:    domain = FactoryGirl.create(:domain)
test/controllers/api/v2/domains_controller_test.rb:    org = FactoryGirl.create(:organization)
```

**Comment by ares** [comment]
Yeah it seems I missed those on last rebase, should be fixed now

**Review by tbrisker — APPROVED** [review]
Test failure is unrelated, merging. Thanks @ares !

**Pull request merged**
